### ISSUE TITLE
Remote notifications message is null when payload uses body_loc_key in foreground

### ISSUE THREAD

**Issue by plus-** [zo0r/react-native-push-notification#1836]
# Bug

**This is a bug on Android.**

I have an issue when trying to internationalize my notifications messages using the `body_loc_key` parameter [See table 2.b](https://firebase.google.com/docs/cloud-messaging/http-server-ref#notification-payload-support)

When the app is in foreground the notification message property is null. In background the notification will display properly and show the value of the resources string.

Using the `body` parameter will properly work in background and foreground (message is the body value)

## Environment info

`    "react-native-push-notification": "^7.0.0",
`

`react-native info` output:

```bash
info Fetching system and libraries information...
(node:22905) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
System:
    OS: macOS 11.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 462.41 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.1 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2, 29.0.3
      System Images: android-R | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 12.3/12C33 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_222 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.4 => 0.63.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found```

**Comment by Dallas62** [comment]
Hi @plus- 
Unfortunately, this repository is not for iOS.

Please refer to:
https://github.com/react-native-push-notification-ios/push-notification-ios

Also note that the iOS library is not using Firebase.

Regards,

**Comment by plus-** [comment]
@Dallas62 this is an issue with Android. I have no problem with IOS. See table 2.b is for Android

**Comment by Dallas62** [comment]
Well, since it's in the `notification` part of the notification, it's not managed by the library but the Firebase SDK, you should find more information in StackoverFlow or Firebase documentation.

**Comment by plus-** [comment]
thanks for reopening this issue, I believe [this](https://github.com/zo0r/react-native-push-notification/blob/60883d102caa4c29668d4c0bcbe885a7a9516b8b/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java#L50) could be extended with 

https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/RemoteMessage.Notification#getBodyLocalizationKey()

If this is not a priority for you to add I can understand and will see if I can provide a PR. Let me know.

**Comment by Dallas62** [comment]
The issue is coming from:
https://github.com/zo0r/react-native-push-notification/blob/master/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java#L50

Which is using the `getBody` and doesn't provide a fallback to the localisation key. I don't know why Firebase makes us reinvent the wheel each time the application is open...

`getTitle` is also impacted.

https://stackoverflow.com/questions/7493287/android-how-do-i-get-string-from-resources-using-its-name
The second answer should work in our case.

I will try to look at it this week-end, if you have time, feel free to make a PR 😉 

Regards,

**Comment by plus-** [comment]
I did a PR, I don't have a Java environment right now, so I couldn't test it.

**Comment by Dallas62** [comment]
Thanks for the PR,

If you have your project, you can apply your change directly in the node_modules, it's not perfect but it's help to test 😉

I will look this weekend and release it ASAP.

**Comment by Dallas62** [comment]
Hi @plus- 
Your patch has been tested and released 😉 (#1837)
Thanks for your report and contribution !
Regards
